[Ang Koon Hwee] iP

angkoonhwee opened the pull request:
## Duke v0.1
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html))

Duke v0.1 frees up your mind of having to remember things you need to do while simultaneously being a chatbot with personality.
* text-based
* easy to learn
* FAST to use
* Replies with emoticons (≧◡≦)

All you need to do to use this application is,
1. download it the [jar file here](https://github.com/angkoonhwee/ip/releases/tag/v0.1)
2. Open the jar file
3. Start using it by adding your tasks!
4. Talk to Duke v0.1 and let it manage your tasks for you

And it is **FREE**! 🐑 

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (Coming soon)

<br />

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
    new Duke(PATHNAME).run();
}
````

Commit:
Add Gradle support

angkoonhwee committed:
level-1

angkoonhwee committed:
level-2

angkoonhwee committed:
level-3

angkoonhwee committed:
Refactor code

angkoonhwee committed:
level 4

angkoonhwee committed:
A-TextUiTesting

angkoonhwee committed:
Level-5 error handling

angkoonhwee committed:
Level 6 Delete

angkoonhwee committed:
incomplete level 7

angkoonhwee committed:
Add level 7

angkoonhwee committed:
Merge pull request #1 from angkoonhwee/branch-Level-7

Merge branch "branch-Level-7" to master

angkoonhwee committed:
Add level 8

angkoonhwee committed:
Merge pull request #2 from angkoonhwee/branch-Level-8

Add level 8

angkoonhwee committed:
add OOP

angkoonhwee committed:
organise into packages

angkoonhwee committed:
Rename List to TaskList

angkoonhwee committed:
Add unit testing

angkoonhwee committed:
Add jar file

angkoonhwee committed:
Add javadocs

angkoonhwee committed:
Merge pull request #3 from angkoonhwee/branch-A-JavaDoc

Add javadocs

angkoonhwee committed:
folloe coding standards

angkoonhwee committed:
Merge pull request #4 from angkoonhwee/branch-A-CodingStandard

merge a-codingstandards

angkoonhwee committed:
Add find command

angkoonhwee committed:
Merge pull request #5 from angkoonhwee/branch-Level-9

Add find command

angkoonhwee committed:
Merge pull request #6 from angkoonhwee/add-gradle-support

Add Gradle support

bnjmnt4n commented on data/duke.txt:
```diff
@@ -0,0 +1,5 @@
+T|0|iP
```
Maybe you should add this to the `.gitignore` so you don't upload your own personal data online.

angkoonhwee replied:
Oh right, I would add it to `.gitignore` file thank you

bnjmnt4n commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,119 @@
+package duke;
+
+import duke.command.*;
```
I think the recommended coding standard is to avoid * imports.

angkoonhwee replied:
Yes I will reformat it thank you for reminding!

bnjmnt4n commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,86 @@
+
+
+    /**
+     * saves the current list to the chat history file
```
Maybe you should modify your JavaDoc to a sentence structure with `Uppercase lines.` for consistency.

bnjmnt4n commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,85 @@
+     * 
+     */
+    public ArrayList<Task> findKeyword(String keyword) {
+        ArrayList<Task> matched = new ArrayList<>();
```
This could be named `matchedTasks` or something else to better convey that it stores multiple items.

bnjmnt4n reviewed (commented):
LGTM, I think your code follows the coding standard very well, except for maybe a lack of consistency in the JavaDoc casing.

angkoonhwee committed:
Modified to CheckStyle standards

EssWhyy commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,43 @@
+
+    @Override
+    public String toSaveString() {
+        return "E|" + super.toSaveString() + "|" + at.format(saveFormatter);
```
I like how you used another string formatter for your tasks when saving to storage, allowing for easier readability when re-loading.

EssWhyy commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,181 @@
+    private static final String END =
+            "    ---------------------------------------------------------------------------------";
+
+    private static final String INDENTATION = "     ";
```
Is there a reason for the indentation for each output line? I assume it is to improve readability?

EssWhyy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,128 @@
+        parsed = input.split(" ");
+        try {
+            switch (parsed[0]) {
+            case ByeCommand.COMMAND:
```
Perhaps you could use a set of enums with the Command names instead? Like LIST, DONE, DELETE. This way there would be no need to include/access the COMMAND constant in each command.

EssWhyy reviewed (commented):
The app is generally well coded, no major coding errors, adequate use of javadocs and jUnit tests.

angkoonhwee committed:
Add GUI

angkoonhwee committed:
Merge pull request #7 from angkoonhwee/branch-Level-10

Add GUI

angkoonhwee committed:
Add Assertions
Methods that processes user input: No checks for empty input

The methods that processes user input has some assumptions made (i.e.
no cases to manage empty user input).

Add assertions at the start of these methods to check for empty strings

Assertions can be used to define assumptions about the program state so
so that the runtime can verify them and aid in debugging.

angkoonhwee committed:
Merge pull request #8 from angkoonhwee/branch-Assertions

Add Assertions

angkoonhwee committed:
Edit Code in accordance to CodeQuality guidelines
Some sections of code does not adhere to the guidelines.

By adhering to give code quality guidelines allows for easier
readability.

Split up long expressions, checked through code to ensure SLAP
and other guidelines.

Edited based on CS2103T guidelines for coursemates to understand the
code better.

Refer to this for the CodeQuality guidelines:
https://nus-cs2103-ay2122s1.github.io/website/schedule/week5/topics.htm
l#w5-4-code-quality

angkoonhwee committed:
Merge pull request #9 from angkoonhwee/branch-CodeQuality

Edit Code in accordance to CodeQuality guidelines

angkoonhwee committed:
D-Client draft

angkoonhwee committed:
Added undo function

angkoonhwee committed:
Merge pull request #10 from angkoonhwee/branch-C-Undo

Branch c undo

angkoonhwee committed:
edit wild imports

angkoonhwee committed:
Add Snooze function

angkoonhwee committed:
Merge pull request #12 from angkoonhwee/branch-B-Snooze

Add Snooze function

angkoonhwee committed:
Improve GUI

angkoonhwee committed:
Add User Guide

angkoonhwee committed:
Update UG images

angkoonhwee committed:
UG image update

angkoonhwee committed:
Update UG images

Xiaoyunnn commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,32 @@
+
+    /**
+     * Constructor for ByeCommand
+     * will mark isExit as true so duke can escape while loop
```
Perhaps you should add a full stop after every sentence.
I think your second sentence should start with a singular verb `Marks` as well.

Xiaoyunnn commented on src/main/java/duke/command/ByeCommand.java:
```diff
@@ -0,0 +1,32 @@
+    /**
+     * Execute ByeCommand, returns a goodbye message from duke
+     *
+     * @param taskList current task list
+     * @param rf Response formatter
+     * @param storage current storage
+     * @return
```
The description after every param should start with a capital letter if I rmb correctly.
I think you forgot to add a description for your return as well XD

Xiaoyunnn commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,42 @@
+    protected boolean isExit = false;
+
+    /**
+     * get the exit check
```
same for this javadoc, should start with caps and it should be a singular verb, i.e. `Gets`

Xiaoyunnn commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,66 @@
+     * @param storage current storage
+     * @throws IOException when there is file save error
+     *
+     * @return
```
another missing description for return XDD

Xiaoyunnn commented on src/main/java/duke/command/ErrorCommand.java:
```diff
@@ -0,0 +1,52 @@
+    /**
+     * Executes Error Command to returns error response
+     *
+     *  @param taskList current list
```
formatting issue? the extra space.

Xiaoyunnn commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,151 @@
+/**
+ * Parser object that parses all the input from user to commands understood by Duke
+ */
+public class Parser {
```
Perhaps you can add javadoc for the public methods in this class since they seem to be non-trivial?

Xiaoyunnn commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,151 @@
+            ArrayIndexOutOfBoundsException,
+            StringIndexOutOfBoundsException {
+        String[] eventDetails = new String[2];
+        String desc = command.split(" /at ")[0].substring(6);
```
According to the code quality, you should minimise the usage of magic numbers like `6` in this case. Perhaps you can declare a constant in this class and change `6` to something like `START_INDEX_EVENT = EventCommand.COMMAND.length() + 1`? same for the other parse methods

Xiaoyunnn reviewed (commented):
LGTM! Just some nits for Javadoc :DDD

angkoonhwee committed:
Refactor code according to Code Quality Guidelines
